launcher/minecraft/Rule.cpp

```diff
@@ -80,14 +113,38 @@ QJsonObject OsRule::toJson()
     QJsonObject ruleObj;
     ruleObj.insert("action", m_result == Allow ? QString("allow") : QString("disallow"));
     QJsonObject osObj;
-    {
-        osObj.insert("name", OpSys_toString(m_system));
-        if(!m_version_regexp.isEmpty())
-        {
-            osObj.insert("version", m_version_regexp);
-        }
+    if(!m_system.isEmpty()) {
+        osObj.insert("name", m_system);
     }
+
     ruleObj.insert("os", osObj);
     return ruleObj;
 }
 
+bool OsRule::applies(Library *parent, const SettingsObjectPtr& settingsObjJavaArch)
+{
+    QString sys;
+    QString arch;
+    if(m_system.contains("-"))
+    {
+        auto parts = m_system.split("-");
+        sys = parts[0];
+        arch = parts[1];
+    }
+    else
+    {
+        sys = m_system;
+    }
+    bool systemCorrect;
+    bool archCorrect = true;
+    systemCorrect = sys == SysInfo::currentSystem();
+    if(!arch.isEmpty())
+    {
+        // if the arch isn't empty then it has to be archDependent right?
+        parent->setArchDependent(true);
+        archCorrect = arch == SysInfo::currentArch(settingsObjJavaArch);
```

Comment:
I don't really like that we are carrying the instance settings all the way from launching to here. Maybe we could create a new data structure instead (I propose calling it `LaunchContext`) that carries data, such as java version